### PR TITLE
Cache Spack for GHA

### PR DESCRIPTION
**Pull request by schmitts**
- make spack ci run use arbor version information
- Cache Spack for GHA

### CODE REVIEW

File: scripts/build_spack_package.sh [hunk 1]
```diff
@@ -46,12 +46,20 @@ case $SPACK_VERSION in
 		echo "SPACK_VERSION" must be \"latest_release\" or \"develop\"
 		exit 1
 esac
+
+mkdir ~/.spack
```
Comment (bcumming) [inline scripts/build_spack_package.sh:50]: Should this be the path `~/.spack-ci` referenced in the other files?

Comment (schmitts) [reply scripts/build_spack_package.sh:50]: No, `~/.spack` is the transient configuration. `~/.spack-ci` is restored from the cache. We should probably rename it to `~/.spack-cache`.

Comment (bcumming) [reply scripts/build_spack_package.sh:50]: Thanks. `spack-cache` is a better name.

**Review by bcumming — APPROVED** [review]
Looks good, thanks!

**Pull request merged**